mailpoet/lib/Automation/Engine/Data/Automation.php

```diff
@@ -171,13 +174,46 @@ public function toArray(): array {
           return $step->toArray();
         }, $this->steps)
       ),
+      'meta' => Json::encode($this->meta),
     ];
   }
 
   private function setUpdatedAt(): void {
     $this->updatedAt = new DateTimeImmutable();
   }
 
+  /**
+   * @param string $key
+   * @return mixed|null
+   */
+  public function getMeta(string $key) {
+    return $this->meta[$key] ?? null;
+  }
+
+  public function getMetas(): array {
```

Comment:
The naming of these methods reveals that `meta` is not a great name 😄
But if you want to keep it for consistency with post meta, then maybe we could rename the plural methods to `getAllMeta`, `deleteAllMeta`, etc.

Reply:
0e115020f